apps/condo/domains/property/components/panels/Builder/MapConstructor.ts

```diff
@@ -567,10 +567,39 @@ class MapEdit extends MapView {
         this.notifyUpdater()
     }
 
+    private getNextUnit (id: string): BuildingUnit {
+        const units = this.map.sections.map(section => [...section.floors].reverse().map(floor => floor.units)).flat(2)
+        const unitIndex = units.findIndex(unit => unit.id === id)
+        const nextIndex = unitIndex + 1
+        return units[nextIndex] || null
+    }
+
+
+    private removeFloor (sectionIdx: number, floorIndex: number): void {
+        const floorToRemove = this.map.sections[sectionIdx].floors[floorIndex]
+        this.map.sections[sectionIdx].floors.splice(floorIndex, 1)
+        this.map.sections[sectionIdx].floors.map(floor => {
+            if (floorToRemove.index < floor.index){
+                floor.index--
+                floor.name = floor.index.toString()
+            }
+            return floor
+        })
+    }
+
     public removeUnit (id: string): void {
         const unitIndex = this.getUnitIndex(id)
+        const nextUnit = this.getNextUnit(id)
         if (unitIndex.unit !== -1) {
-            this.map.sections[unitIndex.section].floors[unitIndex.floor].units.splice(unitIndex.unit, 1)
+            const floorUnits = this.map.sections[unitIndex.section].floors[unitIndex.floor].units
```

Comment:
is it possible to get undefined at `section` and `floor` ?

Reply:
No. 
unitIndex.unit - will be set not to -1 - only if section and floor exist